### PR TITLE
Implement sha1(sha1($pass).$salt) kernel.

### PR DESCRIPTION
**Pull request by cfreal**
We needed `sha1(sha1($pass).$salt)` for a security audit. Built it using 4500 and 4710. The tests run fine.

Thanks for your amazing work.

### CODE REVIEW

**Comment by jsteube** [comment]
This looks nice, but the hash-mode number is incorrect. Could you please rename to 4510?

**Comment by philsmd** [comment]
Could you please also update the files `docs/changes.txt` and `docs/readme.txt` (similar to other commits adding new hash types).
Thank you very much

File: OpenCL/m04510_a1-optimized.cl [hunk 1]
```diff
@@ -0,0 +1,2162 @@
...
+    we_t = hc_rotl32 ((wb_t ^ w6_t ^ w0_t ^ we_t), 1u);
+    SHA1_STEP (SHA1_F1, c, d, e, a, b, we_t);
+    wf_t = hc_rotl32 ((wc_t ^ w7_t ^ w1_t ^ wf_t), 1u);
+    SHA1_STEP (SHA1_F1, b, c, d, e, a, wf_t);
+
+    COMPARE_M_SIMD (d, e, c, b);
+  }
+}
+
+KERNEL_FQ void m04720_m08 (KERN_ATTR_BASIC ())
```
Comment (philsmd) [inline OpenCL/m04510_a1-optimized.cl]: should be m04510_m08

same holds for m04510_m16 , m04510_s04, m04510_s08, m04510_s16 etc

File: OpenCL/m04510_a3-pure.cl [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+#define uint_to_hex_lower8_le(i) make_u32x (l_bin2asc[(i).s0], l_bin2asc[(i).s1])
+#elif VECT_SIZE == 4
+#define uint_to_hex_lower8_le(i) make_u32x (l_bin2asc[(i).s0], l_bin2asc[(i).s1], l_bin2asc[(i).s2], l_bin2asc[(i).s3])
+#elif VECT_SIZE == 8
+#define uint_to_hex_lower8_le(i) make_u32x (l_bin2asc[(i).s0], l_bin2asc[(i).s1], l_bin2asc[(i).s2], l_bin2asc[(i).s3], l_bin2asc[(i).s4], l_bin2asc[(i).s5], l_bin2asc[(i).s6], l_bin2asc[(i).s7])
+#elif VECT_SIZE == 16
+#define uint_to_hex_lower8_le(i) make_u32x (l_bin2asc[(i).s0], l_bin2asc[(i).s1], l_bin2asc[(i).s2], l_bin2asc[(i).s3], l_bin2asc[(i).s4], l_bin2asc[(i).s5], l_bin2asc[(i).s6], l_bin2asc[(i).s7], l_bin2asc[(i).s8], l_bin2asc[(i).s9], l_bin2asc[(i).sa], l_bin2asc[(i).sb], l_bin2asc[(i).sc], l_bin2asc[(i).sd], l_bin2asc[(i).se], l_bin2asc[(i).sf])
+#endif
+
+KERNEL_FQ void m04500_mxx (KERN_ATTR_VECTOR ())
```
Comment (philsmd) [inline OpenCL/m04510_a3-pure.cl]: m04510_mxx ?

**Comment by jsteube** [comment]
Many thanks!

**Pull request merged**